single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh

```diff
@@ -0,0 +1,323 @@
+#!/usr/bin/env bash
+
+DEPLOYMENT_PREFIX=${DEPLOYMENT_PREFIX:-}
+AZURE_SUBSCRIPTION_ID=${AZURE_SUBSCRIPTION_ID:-}
+AZURE_RESOURCE_GROUP_NAME=${AZURE_RESOURCE_GROUP_NAME:-}
+AZURE_RESOURCE_GROUP_LOCATION=${AZURE_RESOURCE_GROUP_LOCATION:-}
+
+if [[ -z "$DEPLOYMENT_PREFIX" ]]; then
+    echo "No deployment prefix [DEPLOYMENT_PREFIX] specified."
+    exit 1
+fi
+if [[ -z "$AZURE_SUBSCRIPTION_ID" ]]; then
+    echo "No Azure subscription id [AZURE_SUBSCRIPTION_ID] specified."
+    exit 1
+fi
+if [[ -z "$AZURE_RESOURCE_GROUP_NAME" ]]; then
+    echo "No Azure resource group [AZURE_RESOURCE_GROUP_NAME] specified."
+    exit 1
+fi
+if [[ -z "$AZURE_RESOURCE_GROUP_LOCATION" ]]; then
+    echo "No Azure resource group [AZURE_RESOURCE_GROUP_LOCATION] specified."
+    exit 1
+fi
+
+# Login to Azure and select the subscription
+if ! AZURE_USERNAME=$(az account show --query user.name); then
+    echo "No Azure account logged in, now trying to log in."
+    az login --output none
+    az account set --subscription "$AZURE_SUBSCRIPTION_ID"
+else
+    echo "Logged in as $AZURE_USERNAME, set the active subscription to \"$AZURE_SUBSCRIPTION_ID\""
+    az account set --subscription "$AZURE_SUBSCRIPTION_ID"
+fi
+
+# Check the resource group and region
+RG_EXISTS=$(az group exists --resource-group "$AZURE_RESOURCE_GROUP_NAME")
+if [[ $RG_EXISTS == "false" ]]; then
+    echo "Creating resource group $AZURE_RESOURCE_GROUP_NAME in $AZURE_RESOURCE_GROUP_LOCATION."
+    az group create --location "$AZURE_RESOURCE_GROUP_LOCATION" --resource-group "$AZURE_RESOURCE_GROUP_NAME" --output none
+else
+    echo "Resource group $AZURE_RESOURCE_GROUP_NAME exists in $AZURE_RESOURCE_GROUP_LOCATION."
+    RG_LOCATION=$(az group show --resource-group "$AZURE_RESOURCE_GROUP_NAME" --query location)
+    if [[ "$RG_LOCATION" != "\"$AZURE_RESOURCE_GROUP_LOCATION\"" ]]; then
+        echo "Resource group $AZURE_RESOURCE_GROUP_NAME is located in $RG_LOCATION, not \"$AZURE_RESOURCE_GROUP_LOCATION\""
+    fi
+fi
+
+# Validate the ARM templates (Jacob)
+
+securityGroupName="${DEPLOYMENT_PREFIX}nsg01"
+securityGroupLocation="$AZURE_RESOURCE_GROUP_LOCATION"
+
+echo "Validating parameters for Network Security Group..."
+if ! az deployment group validate \
+    --resource-group "$AZURE_RESOURCE_GROUP_NAME" \
+    --template-file ./securitygroup/securitygroup.template.json \
+    --parameters \
+        securityGroupName="$securityGroupName" \
+        securityGroupLocation="$securityGroupLocation" \
+    --output none; then
+    echo "Validation error for Network Security Group, please see the error above."
+    exit 1
+else
+    echo "Network Security Group parameters are valid."
+fi
+
+securityGroupName="${DEPLOYMENT_PREFIX}nsg01"
```

Comment:
already defined

Reply:
Yes, I decided to put all params which use for the resource since I thought it's gonna be so hard to know which is already defined/not unless put all variables at the top.
But do you think we should put all variables at the top or have another idea for it?